File: WKWebView/WKWebViewDemo/WKWebViewDemo/ViewController.m
//
//  ViewController.m
//  WKWebViewDemo
//
//  Created by 刘威振 on 2020/7/31.
//  Copyright © 2020 大刘. All rights reserved.
//

#import "ViewController.h"

@interface ViewController () <UITableViewDelegate, UITableViewDataSource>

@property (nonatomic, strong) NSArray *items;
@property (weak, nonatomic) IBOutlet UITableView *tableView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"JKWWebView Demo";
    [self.tableView registerClass:UITableViewCell.class forCellReuseIdentifier:@"cellID"];
    self.items = @[
        @"进度条",
        @"JS交互"
    ];
}

// MARK: - <UITableViewDelegate, UITableViewDataSource>
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cellID" forIndexPath:indexPath];
    cell.textLabel.text = [self.items objectAtIndex:indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    NSString *text = [self.items objectAtIndex:indexPath.row];
    if ([text isEqualToString:@"进度条"]) {
        
    } else if ([text isEqualToString:@"JS交互"]) {
        
    }
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.items.count;
}

@end

File: Demos/sdks/network/ViewController.swift
//
//  ViewController.swift
//  network
//
//  Created by liuweizhen on 2020/5/5.
//  Copyright © 2020 liuxing8807@126.com. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
    @IBOutlet weak var tableView: UITableView!
    
    lazy var items: [DLNetworkEnum] = DLNetworkEnum.all

    override func viewDidLoad() {
        super.viewDidLoad()
        
        setup()
    }
    
    func setup() {
        self.title = "Network"
        
        self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cellID")
        self.tableView.delegate = self
        self.tableView.dataSource = self
    }
    
    // MARK: UITableViewDelegate, UITableViewDelegate
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.items.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell: UITableViewCell = tableView.dequeueReusableCell(withIdentifier: "cellID", for: indexPath)
        let item: DLNetworkEnum = self.items[indexPath.row]
        cell.textLabel?.text = item.rawValue
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let item: DLNetworkEnum = self.items[indexPath.row]
        switch item {
        case .Thread:
            print("TODO://")
        case .Operation:
            print("TODO://")
        case .Semaphore:
            print("TODO://")
        case .DispathGroup:
            let vc: DispatchGroupVC = DispatchGroupVC.loadFromStoryboard()
            self.navigationController?.pushViewController(vc, animated: true)
        }
    }
}
